lib/sample_rate_converter/sample_rate_converter.c

```diff
@@ -0,0 +1,406 @@
+
+	ctx->conversion_ratio = calculate_conversion_ratio(sample_rate_input, sample_rate_output);
+	ctx->conversion_direction =
+		(sample_rate_input < sample_rate_output) ? CONVERSION_DIR_UP : CONVERSION_DIR_DOWN;
+
+	ctx->filter_type = filter;
+	ret = sample_rate_converter_filter_get(filter, ctx->conversion_ratio,
+					       (void **)&filter_coeffs, &filter_size);
+	if (ret) {
+		LOG_ERR("Failed to get filter (%d)", ret);
+		return ret;
+	} else if (filter_size > CONFIG_SAMPLE_RATE_CONVERTER_MAX_FILTER_SIZE) {
+		LOG_ERR("Filter is larger than max size");
+		return -EINVAL;
+	}
+
+	if (conversion_needs_buffering(ctx, ctx->conversion_ratio)) {
+		LOG_DBG("Conversion needs buffering, start with the input buffer filled");
+		if (IS_ENABLED(CONFIG_SAMPLE_RATE_CONVERTER_BIT_DEPTH_16)) {
+			ctx->input_buf.bytes_in_buf =
+				SAMPLE_RATE_CONVERTER_INPUT_BUFFER_NUMBER_OVERFLOW_SAMPLES *
+				sizeof(uint16_t);
+		} else if (IS_ENABLED(CONFIG_SAMPLE_RATE_CONVERTER_BIT_DEPTH_32)) {
+			ctx->input_buf.bytes_in_buf =
+				SAMPLE_RATE_CONVERTER_INPUT_BUFFER_NUMBER_OVERFLOW_SAMPLES *
+				sizeof(uint32_t);
+		}
+		memset(ctx->input_buf.buf, 0, ctx->input_buf.bytes_in_buf);
+	} else {
+		ctx->input_buf.bytes_in_buf = 0;
+	}
+
+	/* Ring buffer will be used to store spillover output samples */
+	ring_buf_init(&ctx->output_ringbuf, ARRAY_SIZE(ctx->output_ringbuf_data),
+		      ctx->output_ringbuf_data);
+
+#ifdef CONFIG_SAMPLE_RATE_CONVERTER_BIT_DEPTH_16
+	if (ctx->conversion_direction == CONVERSION_DIR_UP) {
+		arm_err = arm_fir_interpolate_init_q15(&ctx->fir_interpolate_q15,
+						       ctx->conversion_ratio, filter_size,
+						       (q15_t *)filter_coeffs, ctx->state_buf_15,
+						       CONFIG_SAMPLE_RATE_CONVERTER_BLOCK_SIZE_MAX);
+
+	} else {
+		arm_err = arm_fir_decimate_init_q15(&ctx->fir_decimate_q15, filter_size,
+						    ctx->conversion_ratio, (q15_t *)filter_coeffs,
+						    ctx->state_buf_15,
+						    CONFIG_SAMPLE_RATE_CONVERTER_BLOCK_SIZE_MAX);
+	}
+#elif CONFIG_SAMPLE_RATE_CONVERTER_BIT_DEPTH_32
+	if (ctx->conversion_direction == CONVERSION_DIR_UP) {
+		arm_err = arm_fir_interpolate_init_q31(&ctx->fir_interpolate_q31,
+						       ctx->conversion_ratio, filter_size,
+						       (q31_t *)filter_coeffs, ctx->state_buf_31,
+						       CONFIG_SAMPLE_RATE_CONVERTER_BLOCK_SIZE_MAX);
+	} else {
+		arm_err = arm_fir_decimate_init_q31(&ctx->fir_decimate_q31, filter_size,
+						    ctx->conversion_ratio, (q31_t *)filter_coeffs,
+						    ctx->state_buf_31,
+						    CONFIG_SAMPLE_RATE_CONVERTER_BLOCK_SIZE_MAX);
+	}
+#endif
+	if (arm_err == ARM_MATH_LENGTH_ERROR) {
+		LOG_ERR("Filter size is not a multiple of conversion ratio");
+		return -EINVAL;
+	} else if (arm_err != ARM_MATH_SUCCESS) {
+		LOG_ERR("Unknown error during interpolator/decimator initialization (%d)", arm_err);
+		return -EINVAL;
+	}
+
+	LOG_INF("Sample rate converter initialized. Input sample rate: %d, Output sample rate: %d, "
```

Comment:
```suggestion
	LOG_DBG("Sample rate converter initialized. Input sample rate: %d, Output sample rate: %d, "
```

Reply:
Haha, team work at it's best!

I'll change it back to debug then, since that was my initial gut feeling as well.